Mixmatch/streetview_dataset/parse_data_to_tfrecord_lib_test.ipynb

```diff
@@ -0,0 +1,239 @@
+    "from PIL import Image  # used to read images from directory\n",
+    "import tensorflow as tf\n",
+    "import os\n",
+    "import io\n",
+    "import IPython.display as display"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Test function img_to_example():\n",
+    "The function reads in image files and outputs tf.examples. The output should store the information feeded into the function."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "IMG_PATH = './TC11/svt1/img/19_00.jpg'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "try:\n",
+    "    img = Image.open(IMG_PATH, \"r\")\n",
+    "except Exception as e:\n",
+    "    print(e)\n",
+    "    print(IMG_PATH + \" is not valid\")\n",
+    "\n",
+    "example = img_to_example(img, label=0)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Testing\n",
+    "# The label feature should be a int64_list with value of {value: 0}\n",
+    "print(example.features.feature['label'])\n",
+    "\n",
+    "# The image feature should be a bytes_list storing the image\n",
+    "print(example.features.feature['image'].SerializeToString())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Function Test: read_tfrecord()\n",
+    "The function should read tfrecord files as input, and return DatasetV1Adapter storing a list of examples"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Global constants\n",
+    "# Information from input tfrecord files\n",
+    "SOURCE_ID = 'image/source_id'\n",
+    "BBOX_CONFIDENCE = 'image/object/bbox/confidence'\n",
+    "BBOX_XMIN = 'image/object/bbox/xmin'\n",
+    "BBOX_YMIN = 'image/object/bbox/ymin'\n",
+    "BBOX_XMAX = 'image/object/bbox/xmax'\n",
+    "BBOX_YMAX = 'image/object/bbox/ymax'\n",
+    "\n",
+    "INPUT_RECORD_DIR = './streetlearn-detections/'\n",
+    "file_name = \"./streetlearn_detections_tfexample-00000-of-01000.tfrecord\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "parsed_image_dataset = read_tfrecord(os.path.join(INPUT_RECORD_DIR, file_name))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Testing\n",
+    "# Check the data in the parsed_image_dataset\n",
+    "for example in parsed_image_dataset.take(1):\n",
+    "    print(type(example[SOURCE_ID]))\n",
+    "    print(example[BBOX_CONFIDENCE])\n",
+    "    print(example[BBOX_XMIN], example[BBOX_YMIN], example[BBOX_XMAX], example[BBOX_YMAX])"
```

Comment:
Similar to above, please compare it with the expected value instead of printing the result if possible.

Reply:
Done. Compared by asserting to the ground truth.